src/v/storage/segment_utils.cc

```diff
@@ -853,13 +858,44 @@ make_concatenated_segment(
           });
         return seg_it->index().broker_timestamp();
     }();
+
+    // If both of the segments have a clean_compact_timestamp set, then the new
+    // index should use the maximum timestamp. If at least one segment doesn't
+    // have a clean_compact_timestamp, then the new index should not either.
+    auto new_clean_compact_timestamp =
+      [&]() -> std::optional<model::timestamp> {
+        // invariants: segments is not empty, but for completeness handle the
+        // empty case
+        if (unlikely(segments.empty())) {
+            return std::nullopt;
+        }
+        std::optional<model::timestamp> new_ts;
+        for (const auto& seg : segments) {
```

Comment:
why not use `new_broker_timestamp` similar to how new_broker_timestamp is computed above?